src/sentry/snuba/spans_rpc.py

```diff
@@ -284,10 +295,14 @@ def run_top_events_timeseries_query(
         remapped_groupby = {}
         # Remap internal attrs back to public ones
         for col in groupby_columns:
-            resolved_groupby, _ = search_resolver.resolve_attribute(col)
-            remapped_groupby[resolved_groupby.public_alias] = groupby_attributes[
-                resolved_groupby.internal_name
-            ]
+            if col in ["project", "project.slug"]:
+                resolved_groupby, _ = search_resolver.resolve_attribute("project.id")
+                remapped_groupby[col] = params.project_id_map[
+                    int(groupby_attributes[resolved_groupby.internal_name])
+                ]
```

Comment:
Same here, hard coding it like this will encourage this pattern in the future which we should try to avoid.